lib/addons/efs-csi-driver/index.ts

```diff
@@ -0,0 +1,82 @@
+import { Construct } from "@aws-cdk/core";
+import { ClusterInfo } from "../../spi";
+import { HelmAddOn, HelmAddOnUserProps } from "../helm-addon";
+import { EfsDriverPolicyDocument } from "./iam-policy";
+import { registries }  from "../../utils/registry-utils";
+import * as iam from "@aws-cdk/aws-iam";
+
+const EFS_CSI_DRIVER = "aws-efs-csi-driver";
+const EFS_CSI_CONTROLLER_SA = "efs-csi-controller-sa";
+const EFS_REGISTRY_SUFFIX = "eks/aws-efs-csi-driver";
+
+/**
+ * Configuration options for the add-on.
+ */
+export interface EfsCsiDriverProps extends HelmAddOnUserProps {
+    /**
+     * Version of the driver to deploy. Uses chart version 2.2.3 by default if this value is not provided
+     */
+    version?: string,
+    /***
+     * Number of replicas to be deployed. If not provided, it defaults to 2. Note that the number of replicas
+     * should be less than or equal to the number of nodes in the cluster otherwise some
+     * pods will be left of pending state
+     */
+    replicaCount?: number
+
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps: EfsCsiDriverProps = {
+    version: '2.2.4',
+    namespace: "kube-system",
+    repository: "https://kubernetes-sigs.github.io/aws-efs-csi-driver/",
+    name: EFS_CSI_DRIVER,
+    chart: EFS_CSI_DRIVER,
+    replicaCount: 2
+};
+
+export class EfsCsiDriverAddOn extends HelmAddOn {
+
+    readonly options: EfsCsiDriverProps;
+
+    constructor(props?: EfsCsiDriverProps) {
+        super({ ...defaultProps as any, ...props });
+        this.options = this.props as EfsCsiDriverProps;
+    }
+
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        // Create service account and policy
+        const cluster = clusterInfo.cluster;
+        const serviceAccount = cluster.addServiceAccount(EFS_CSI_CONTROLLER_SA, {
+            name: EFS_CSI_CONTROLLER_SA,
+            namespace: this.options.namespace,
+        });
+        EfsDriverPolicyDocument().Statement.forEach((statement) => {
+            serviceAccount.addToPrincipalPolicy(iam.PolicyStatement.fromJson(statement));
+        });
+
+        // Lookup appropriate image repo
+        const repo = registries.get(clusterInfo.cluster.stack.region) + EFS_REGISTRY_SUFFIX;
+
+        // Define chart
+        const efsCsiDriverChart = this.addHelmChart(clusterInfo, {
+            clusterName: cluster.clusterName,
```

Comment:
Since the generic helm props allow passing values from the user, we generally merge whatever user passed with the values that we generated, with user passed value taking precedence. 

I observe quite a bit of options here: https://github.com/kubernetes-sigs/aws-efs-csi-driver

Example of value merging is for example here: https://github.com/aws-quickstart/ssp-amazon-eks/blob/main/lib/addons/appmesh/index.ts#L100

Reply:
Oh....yeah.....my bad, I' fix it